src/engraving/dom/edit.cpp

```diff
@@ -763,6 +763,16 @@ TextBase* Score::addText(TextStyleType type, EngravingItem* destinationElement)
         chordRest->undoAddAnnotation(textBox);
         break;
     }
+    case TextStyleType::DYNAMICS: {
+        ChordRest* chordRest = chordOrRest(destinationElement);
+        if (!chordRest) {
+            break;
+        }
+        textBox = Factory::createDynamic(dummy()->segment());
+        chordRest->undoAddAnnotation(textBox);
+        textBox->setFlag(ElementFlag::IS_PREVIEW, true);
```

Comment:
I think it would be better to set this flag in the popup code. The popup is namely the only thing that will create a preview dynamic, so it would be good to keep the code that sets the flag close to the code that creates the preview dynamic. 

(Also, when splitting this PR into smaller PRs, you'll need to do that anyway, because in the PR about the popup, this code won't exist yet 🙂)